packages/components/src/circular-option-picker/types.ts

```diff
@@ -41,21 +42,54 @@ export type CircularOptionPickerProps = {
 	 */
 	children?: ReactNode;
 	/**
-	 * Whether the keyboard interaction should wrap around.
+	 * Whether the control should present as a set of buttons,
+	 * each with its own tab stop.
+	 *
+	 * @default false
+	 */
+	asButtons?: boolean;
+};
+
+type WithBaseId = {
+	baseId: string;
+};
+
+type FullListboxCircularOptionPickerProps = CommonCircularOptionPickerProps & {
+	/**
+	 * Prevents keyboard interaction from wrapping around.
+	 * Only used when `asButtons` is not true.
 	 *
 	 * @default true
 	 */
 	loop?: boolean;
 } & (
-	| {
-			'aria-label': string;
-			'aria-labelledby'?: never;
-	  }
-	| {
-			'aria-label'?: never;
-			'aria-labelledby': string;
-	  }
- );
+		| {
+				'aria-label': string;
+				'aria-labelledby'?: never;
+		  }
+		| {
+				'aria-label'?: never;
+				'aria-labelledby': string;
+		  }
+	 );
+
+export type ListboxCircularOptionPickerProps = WithBaseId &
+	Omit<
+		FullListboxCircularOptionPickerProps,
+		'asButtons' | 'actions' | 'options'
+	>;
+
+type FullButtonsCircularOptionPickerProps = CommonCircularOptionPickerProps;
+
+export type ButtonsCircularOptionPickerProps = WithBaseId &
+	Omit<
+		FullButtonsCircularOptionPickerProps,
+		'asButtons' | 'actions' | 'options'
+	>;
+
+export type CircularOptionPickerProps =
+	| ( { asButtons?: false } & FullListboxCircularOptionPickerProps )
+	| ( { asButtons: true } & FullButtonsCircularOptionPickerProps );
```

Comment:
Let's refactor the way `asButtons` is defined:

```diff
diff --git a/packages/components/src/circular-option-picker/types.ts b/packages/components/src/circular-option-picker/types.ts
index 2c2b92ba09..463e519f57 100644
--- a/packages/components/src/circular-option-picker/types.ts
+++ b/packages/components/src/circular-option-picker/types.ts
@@ -41,13 +41,6 @@ type CommonCircularOptionPickerProps = {
 	 * The child elements.
 	 */
 	children?: ReactNode;
-	/**
-	 * Whether the control should present as a set of buttons,
-	 * each with its own tab stop.
-	 *
-	 * @default false
-	 */
-	asButtons?: boolean;
 };
 
 type WithBaseId = {
@@ -55,6 +48,13 @@ type WithBaseId = {
 };
 
 type FullListboxCircularOptionPickerProps = CommonCircularOptionPickerProps & {
+	/**
+	 * Whether the control should present as a set of buttons,
+	 * each with its own tab stop.
+	 *
+	 * @default false
+	 */
+	asButtons?: false;
 	/**
 	 * Prevents keyboard interaction from wrapping around.
 	 * Only used when `asButtons` is not true.
@@ -79,7 +79,15 @@ export type ListboxCircularOptionPickerProps = WithBaseId &
 		'asButtons' | 'actions' | 'options'
 	>;
 
-type FullButtonsCircularOptionPickerProps = CommonCircularOptionPickerProps;
+type FullButtonsCircularOptionPickerProps = CommonCircularOptionPickerProps & {
+	/**
+	 * Whether the control should present as a set of buttons,
+	 * each with its own tab stop.
+	 *
+	 * @default false
+	 */
+	asButtons: true;
+};
 
 export type ButtonsCircularOptionPickerProps = WithBaseId &
 	Omit<
@@ -88,8 +96,8 @@ export type ButtonsCircularOptionPickerProps = WithBaseId &
 	>;
 
 export type CircularOptionPickerProps =
-	| ( { asButtons?: false } & FullListboxCircularOptionPickerProps )
-	| ( { asButtons: true } & FullButtonsCircularOptionPickerProps );
+	| FullListboxCircularOptionPickerProps
+	| FullButtonsCircularOptionPickerProps;
 
 export type DropdownLinkActionProps = {
 	buttonProps?: Omit<
```

This way has, IMO, a couple of advantages:
- the `FullListboxCircularOptionPickerProps` and `FullButtonsCircularOptionPickerProps` types are _actually_ representing the full list of props
- the JSDocs for the `asButtons` prop are correctly parsed by intellisense